ansible/README.md

```diff
@@ -0,0 +1,121 @@
+Deploying Openwhisk using Ansible
+=========
+
+
+### Getting started
+
+If you want to deploy Openwhisk locally using Ansible, you first need to install Ansible on your development environment:
+
+#### Ubuntu users
+```
+sudo apt-get install python-pip
+sudo pip install ansible==2.0.2.0
+```
+
+#### Vagrant users
+Nothing to be done, Ansible is already installed during vagrant provisioning.
+
+#### Mac users
+It is assumed that a VM has been provisioned using **docker-machine**.
+
+```
+brew install python
+pip install ansible==2.0.2.0
+
+cd ansible
+ansible-playbook setup.yml
+```
+
+After this there should be a `hosts` file in the `ansible/environments/mac` directory.
+
+To verify the hosts file you can do a quick ping to the docker machine:
+
+```
+cd ansible
+ansible all -i environments/mac -m ping
+```
+
+Should result in something like:
+
+```
+ansible | SUCCESS => {
+    "changed": false,
+    "ping": "pong"
+}
+192.168.99.100 | SUCCESS => {
+    "changed": false,
+    "ping": "pong"
+}
+```
+
+### Using Ansible
+**Caveat:** All ansible commands are meant to be executed from the `ansible` directory.
+This is important because that's where `ansible.cfg` is located which contains generic settings that are needed for the remaining steps.
+
+In all instructions, replace `<environment>` with your target environment. e.g. `mac`if you want to deploy using a local mac setup.
+
+In all instructions, replace `<openwhisk_home>` with the base directory of your Openwhisk source tree. e.g. `openwhisk`
+
+
+#### Install Prerequisites
+This needs to be done only once per environment. It will install necessary prerequisites on all target hosts in the environment.
+
+```
+ansible-playbook -i environments/<environment> prereq.yml
+```
+
+**Caveat:** Mac users who have a docker-machine setup will have to re-run this playbook every time the boot2docker machine is rebooted.
+This is because installed prereqs are not persisted on TinyCore Linux.
+
+
+### Deploying Using CouchDB
+- Uncomment the couchdb secion in `environments/<environment>/group_vars/all`
+- Comment out the cloudant section
+- Then execute
+
+```
+ansible-playbook -i environments/<environment> couchdb.yml
+ansible-playbook -i environments/<environment> couchdb.yml -e mode=initdb
```

Comment:
On that matter I actually opened an issue with the invoker: https://github.com/openwhisk/openwhisk/issues/198

there is two sides to the invoker life cycle:

1. planned outage (SIGTERM, i.e. hotswap, redeployment, manual restart etc.)
2. unplanned outage (SIGKILL, i.e.invoker dies, VM dies, docker dies, everybody dies)

currently we only cater to the second problem with our RAS efforts (restart=always policy etc.)

In the case of hotswap, i.e. deliberate shutdown of the invoker it should do some sort of clean up to allow me to get a clean system afterwards. I proposed in #198 that the invoker do a garbage collection on SIGTERM and unpause + remove all remaining user containers before it shuts itself down. that would allow us to remove the "docker rm -f" line from the playbook.